File: MEMO.DAL/Entities/Dictionary.cs
using MEMO.DAL.Interfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace MEMO.DAL.Entities
{
    public class Dictionary : IAuditable
    {
        public Guid Id { get; set; }
        [MaxLength(20)]
        public string Name { get; set; }
        [MaxLength(250)]
        public string Description { get; set; }
        public bool IsPublic { get; set; }
        public bool IsFastAccessible { get; set; }

        [NotMapped]
        public int TranslationCount { get; set; }

        [NotMapped]
        public int ViewerCount { get; set; }

        public ICollection<Translation> Translations { get; set; } = new List<Translation>();

        [MinLength(1)]
        public ICollection<UserDictionary> UserDictionaries { get; set; } = new List<UserDictionary>();

        [MaxLength(2)]
        public ICollection<DictionaryLanguage> DictionaryLanguages { get; set; } = new List<DictionaryLanguage>();
        
        public DateTime CreationDate { get; set; }
    }
}

File: MEMO.MobilClient/app/src/main/java/hu/mobilclient/memo/model/memoapi/Dictionary.kt
package hu.mobilclient.memo.model.memoapi

import android.graphics.drawable.Drawable
import com.google.gson.annotations.SerializedName
import hu.mobilclient.memo.App
import hu.mobilclient.memo.R
import hu.mobilclient.memo.helpers.Constants
import hu.mobilclient.memo.helpers.DateParser
import hu.mobilclient.memo.model.enums.LanguageCode
import hu.mobilclient.memo.model.memoapi.interfaces.IAuditable
import java.util.*

class Dictionary(@SerializedName("id")var Id: UUID = UUID(0,0),
                 @SerializedName("name") var Name: String = Constants.EMPTY_STRING,
                 @SerializedName("description") var Description: String = Constants.EMPTY_STRING,
                 @SerializedName("isPublic") var IsPublic: Boolean = false,
                 @SerializedName("isFastAccessible") var IsFastAccessible: Boolean = false,
                 @SerializedName("owner") var Owner: User = User(),
                 @SerializedName("source") var Source: Language = Language(),
                 @SerializedName("destination") var Destination: Language = Language(),
                 @SerializedName("translations") var Translations: MutableList<Translation> = ArrayList(),
                 @SerializedName("viewers") var Viewers: MutableList<User> = ArrayList(),
                 @SerializedName("translationCount")var TranslationCount: Int = Constants.ZERO,
                 @SerializedName("viewerCount")var ViewerCount: Int = Constants.ZERO,
                 @SerializedName("creationDate") override var CreationDate: String = Constants.EMPTY_STRING) : IAuditable {

    fun parsedCreationDate() = DateParser.parse(CreationDate)

    fun isOwn(): Boolean{
        val userId = Owner.Id
        return App.isCurrent(userId)
    }

    val icon: Drawable
        get(){
            return when(Source.Code){
                LanguageCode.EN -> when(Destination.Code){
                    LanguageCode.HU -> App.instance.applicationContext.resources.getDrawable(R.drawable.ic_en_hu_flag, null)
                    LanguageCode.EN -> App.instance.applicationContext.resources.getDrawable(R.drawable.ic_en_flag, null)
                }
                LanguageCode.HU -> when(Destination.Code){
                    LanguageCode.HU -> App.instance.applicationContext.resources.getDrawable(R.drawable.ic_hu_flag, null)
                    LanguageCode.EN -> App.instance.applicationContext.resources.getDrawable(R.drawable.ic_hu_en_flag, null)
                }
            }
        }

    override fun toString() = Name
}